src/Meilisearch/Extensions/HttpExtensions.cs

```diff
@@ -81,13 +83,49 @@ internal static void AddDefaultUserAgent(this HttpClient client)
             client.DefaultRequestHeaders.Add("User-Agent", version.GetQualifiedVersion());
         }
 
-        private static StringContent PrepareJsonPayload<T>(T body, JsonSerializerOptions options = null)
+        private static StringContent PrepareJsonPayload<T>(T body, JsonSerializerOptions? options = null)
         {
             options = options ?? Constants.JsonSerializerOptionsWriteNulls;
             var payload = new StringContent(JsonSerializer.Serialize(body, options), Encoding.UTF8, "application/json");
             payload.Headers.ContentType.CharSet = string.Empty;
 
             return payload;
         }
+
+        private static Task<HttpResponseMessage> PatchAsync(this HttpClient client, string? requestUri, HttpContent content, CancellationToken cancellationToken)
+        {
+            var uri = string.IsNullOrEmpty(requestUri) ? null : new Uri(requestUri, UriKind.RelativeOrAbsolute);
+            return client.PatchAsync(uri, content, cancellationToken);
+        }
+
+        private static Task<HttpResponseMessage> PatchAsync(this HttpClient client, Uri? requestUri, HttpContent content, CancellationToken cancellationToken)
+        {
+            // HttpClient.PatchAsync is not available in .NET standard and NET462
+            var method = new HttpMethod("PATCH");
+            var request = new HttpRequestMessage(method, requestUri) { Content = content };
+            return client.SendAsync(request, cancellationToken);
+        }
+
+        internal static Task<HttpResponseMessage> PatchAsJsonAsync<TValue>(this HttpClient client, string? requestUri, TValue value, JsonSerializerOptions? options = null, CancellationToken cancellationToken = default)
+        {
+            if (client == null)
+            {
+                throw new ArgumentNullException(nameof(client));
+            }
+
+            var content = JsonContent.Create(value, mediaType: null, options);
```

Comment:
I hesitate but this method is used for the custom method we made from  HTTP-`JsonCustomAsync` so I preferred to keep the basic implementation of what the function should be in case one day we will use the real one.
Don't you think?